traffic_ops/traffic_ops_golang/deliveryservice/servers/servers.go

```diff
@@ -97,130 +100,107 @@ func (dss *TODeliveryServiceServer) Validate(db *sqlx.DB) []error {
+		fmt.Fprintf(w, "%s", respBts)
 	}
-	rollbackTransaction = false
-	return nil, tc.NoError
 }
+func (dss *TODeliveryServiceServer) readDSS(db *sqlx.DB, params map[string]string, user auth.CurrentUser) (*tc.DeliveryServiceServerResponse, []error, tc.ApiErrorType) {
+	limitstr := params["limit"]
+	pagestr := params["page"]
+	orderby := params["deliveryService"]
+	limit := 20
+	offset := 1
+	page := 1
+	var err error = nil
+
+	if limitstr != "" {
+		limit, err = strconv.Atoi(limitstr)
 
-func insertQuery() string {
-	query := `INSERT INTO deliveryservice_server (
-deliveryservice,
-server) VALUES (
-:ds_id,
-:server_id) RETURNING deliveryservice, server, last_updated`
-	return query
-}
+		if err != nil {
+			log.Errorf("limit parameter is not an integer")
+			return nil, []error{errors.New("limit parameter must be an integer.")}, tc.SystemError
+		}
+	}
 
-func (dss *TODeliveryServiceServer) Read(db *sqlx.DB, params map[string]string, user auth.CurrentUser) ([]interface{}, []error, tc.ApiErrorType) {
-	idstr, ok := params["id"]
+	if pagestr != "" {
+		offset, err = strconv.Atoi(pagestr)
+		page, err = strconv.Atoi(pagestr)
 
-	if !ok {
-		log.Errorf("Deliveryservice Server Id missing")
-		return nil, []error{errors.New("Deliverservice id is required.")}, tc.DataMissingError
+		if err != nil {
+			log.Errorf("page parameter is not an integer")
+			return nil, []error{errors.New("page parameter must be an integer.")}, tc.SystemError
+		}
+
+		if offset > 0 {
+			offset -= 1
+		}
+
+		offset *= limit
 	}
-	id, err := strconv.Atoi(idstr)
 
-	if err != nil {
-		log.Errorf("Deliveryservice Server Id is not an integer")
-		return nil, []error{errors.New("Deliverservice id is not an integer.")}, tc.SystemError
+	if orderby == "" {
+		orderby = "deliveryService"
 	}
 
-	query := selectQuery()
+	query := selectQuery(orderby, strconv.Itoa(limit), strconv.Itoa(offset))
 	log.Debugln("Query is ", query)
 
-	rows, err := db.Queryx(query, id)
+	rows, err := db.NamedQuery(query, dss)
 	if err != nil {
 		log.Errorf("Error querying DeliveryserviceServers: %v", err)
 		return nil, []error{tc.DBError}, tc.SystemError
 	}
 	defer rows.Close()
 
-	servers := []interface{}{}
+	servers := []tc.DeliveryServiceServer{}
 	for rows.Next() {
-		var s tc.DssServer
+		var s tc.DeliveryServiceServer
 		if err = rows.StructScan(&s); err != nil {
 			log.Errorf("error parsing dss rows: %v", err)
 			return nil, []error{tc.DBError}, tc.SystemError
 		}
-		hiddenField := ""
-		if user.PrivLevel < auth.PrivLevelAdmin {
-			s.ILOPassword = &hiddenField
-		}
 		servers = append(servers, s)
 	}
 
-	return servers, []error{}, tc.NoError
-
+	return &tc.DeliveryServiceServerResponse{orderby, servers,page, limit}, []error{}, tc.NoError
 }
 
-//The Parameter implementation of the Deleter interface
 //all implementations of Deleter should use transactions and return the proper errorType
+
+//The Parameter implementation of the Deleter interface
```

Comment:
Function-level comment should begin with the function name, i.e. `// Delete`, to make Godoc work properly